js/adapt-xapi.js

```diff
@@ -93,10 +105,45 @@ define(function(require) {
         return;
       }
 
-      var statement = statementModel.getStatementObject();
+      this.sendStatement(
+        statementModel.getStatement()
+      );
+    },
+
+    onQuestionAttempt: function(question) {
+      if (!question.model.get('_recordInteraction')) {
+        return;
+      }
+
+      var data = {
+        activityId: this.get('activityId'),
+        actor: this.get('actor'),
+        registration: this.get('registration'),
+        model: question.model
+      };
+
+      var statementModel;
+      switch (question.model.get('_component')) {
+        case "mcq":
```

Comment:
I'm not sure about this as it means very tight coupling between the question components and the xAPI extension. Is it necessary to have specific statement models when question models expose the same properties anyway?  Could we get around requiring specific implementations per question type if we had the xAPI extension decorate the properties of the question model, and let us know what properties we should be passing?